toolkit/tools/pkg/imagecustomizerlib/customizeos.go

```diff
@@ -43,6 +72,17 @@ func doOsCustomizations(ctx context.Context, rc *ResolvedConfig, imageConnection
 		}
 	}
 
+	// If UKI passthrough mode, verify no kernel binaries appeared in /boot
+	if rc.Config.OS.Uki != nil && rc.Config.OS.Uki.Mode == imagecustomizerapi.UkiModePassthrough {
+		hasKernels, err := hasKernelBinariesInBoot(imageChroot.RootDir())
```

Comment:
It might be good to also check for initramfs files.

Reply:
Fixed.